src/Features/Core/Portable/EmbeddedLanguages/StackFrame/StackFrameParser.cs

```diff
@@ -0,0 +1,455 @@
+        }
+
+        /// <summary>
+        /// Type arguments for stacks are only valid on method declarations, and can have either '[' or '&lt;' as the 
+        /// starting character depending on output source.
+        /// 
+        /// ex: MyNamespace.MyClass.MyMethod[T](T t)
+        /// ex: MyNamespace.MyClass.MyMethod&lt;T&lt;(T t)
+        /// 
+        /// Assumes the identifier "MyMethod" has already been parsed, and the type arguments will need to be parsed.
+        /// </summary>
+        private ParseResult<StackFrameTypeArgumentList> TryParseTypeArguments()
+        {
+            if (!_lexer.ScanCurrentCharAsTokenIfMatch(
+                    kind => kind is StackFrameKind.OpenBracketToken or StackFrameKind.LessThanToken,
+                    out var openToken))
+            {
+                return ParseResult<StackFrameTypeArgumentList>.Empty;
+            }
+
+            var closeBrackeKind = openToken.Kind is StackFrameKind.OpenBracketToken
+                ? StackFrameKind.CloseBracketToken
+                : StackFrameKind.GreaterThanToken;
+
+            using var _ = ArrayBuilder<StackFrameNodeOrToken>.GetInstance(out var builder);
+            var currentIdentifier = _lexer.TryScanIdentifier(scanAtTrivia: false, scanLeadingWhitespace: true, scanTrailingWhitespace: true);
+            StackFrameToken closeToken = default;
+
+            while (currentIdentifier.HasValue && currentIdentifier.Value.Kind == StackFrameKind.IdentifierToken)
+            {
+                builder.Add(new StackFrameIdentifierNameNode(currentIdentifier.Value));
+
+                if (_lexer.ScanCurrentCharAsTokenIfMatch(closeBrackeKind, out closeToken))
+                {
+                    break;
+                }
+
+                if (!_lexer.ScanCurrentCharAsTokenIfMatch(StackFrameKind.CommaToken, out var commaToken))
+                {
+                    return ParseResult<StackFrameTypeArgumentList>.Abort;
+                }
+
+                builder.Add(commaToken);
+                currentIdentifier = _lexer.TryScanIdentifier();
+            }
+
+            if (builder.Count == 0)
+            {
+                return ParseResult<StackFrameTypeArgumentList>.Abort;
+            }
+
+            if (closeToken.IsMissing)
+            {
+                return ParseResult<StackFrameTypeArgumentList>.Abort;
+            }
+
+            var separatedList = new EmbeddedSeparatedSyntaxNodeList<StackFrameKind, StackFrameNode, StackFrameIdentifierNameNode>(builder.ToImmutable());
+            return new(new(openToken, separatedList, closeToken));
+        }
+
+        /// <summary>
+        /// MyNamespace.MyClass.MyMethod[|(string s1, string s2, int i1)|]
+        /// Takes parameter declarations from method text and parses them into a <see cref="StackFrameParameterList"/>.
+        /// </summary>
+        private ParseResult<StackFrameParameterList> ParseMethodParameters()
```

Comment:
method parameters are required right?  if so, this should be TryParseMethodParameters.  And null indicates failure.